kpatch-build/kpatch-build

```diff
@@ -1161,8 +1183,9 @@ else
 fi
 
 cd "$TEMPDIR/patch" || die
-if [[ -z "$OOT_MODULE" ]]; then
-	KPATCH_BUILD="$SRCDIR"
+
+if [[ -z "$OOT_MODULE" || ! -e "/lib/modules/$ARCHVERSION/build" ]]; then
+	KPATCH_BUILD="$USERSRCDIR"
```

Comment:
Hi, thanks for the update.

Unfortunately this change breaks the builds that don't specify `$USERSRCDIR`.

Reply:
But, if I understood correctly, that was the whole point of 
> It seems to me that instead of changing `--source` semantics it'll be better to add another option for `OOT_MODULE` case that will define `OOT_MODULE_PATH` or something in that vein. We then'll be able to keep using `SRCDIR` normally

In what cases the build will break?

Reply:
> In what cases the build will break?

Sorry, I should have been more clear. This breaks the normal build (without oot_module) without `USERSRCDIR` In that case `SRCDIR` should be used for kpatch-build.  I think ideally we would keep SRCDIR untouched by OOT_MODULE code and only use OOT_SOURCE directly when needed. Can't tell if that will be easy to do though, we don't normaly build out-of-tree modules so I am not very familiar with that code.

Reply:
Unfortunately, if we do not set SRCDIR to OOT_SOURCE and use it directly, that will add to many if's and that is not good. I tried to mitigate the issue in easier way, please check my last update.

Reply:
Ok, this makes sense.

One last question: does `OOT_MODULE` still need  to depend on `USERSRCDIR`, or can we now decouple it?

Reply:
At what line does `OOT_MODULE` depend on `USERSRCDIR`?

Reply:
> At what line does OOT_MODULE depend on USERSRCDIR?

https://github.com/dynup/kpatch/pull/1234/files#diff-0c4525f1e83472a84e8ac916c7860ab840bb98069cc6bb021f18675f1f3d9092R666 is the main place but there are also others like https://github.com/dynup/kpatch/pull/1234/files#diff-0c4525f1e83472a84e8ac916c7860ab840bb98069cc6bb021f18675f1f3d9092R666 which is inside `if [[ -n "$USERSRCDIR" ]]; then` clause.

Reply:
Well, i don't think it shouldn't depend on `USERSRCDIR`. Where else can we get working Kbuild?

Reply:
Normally (for non-oot cases) if `USERSRCDIR` is not specified kpatch-build would download running distro kernel's sources and prepare them for build in `SRCDIR` (`~/.kpatch/src` by default). There is [this whole section](https://github.com/dynup/kpatch/blob/master/kpatch-build/kpatch-build#L744) handling it. I think that can work for OOT modules as well.